androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/home/HomeScreen.kt

```diff
@@ -1,21 +1,229 @@
+            name = "Avishek",
+            avatarUrl = "https://cataas.com/cat/says/hello%20world!"
+        )
+        surveyUiModels = listOf(
+            SurveyUiModel(
+                id = "1",
+                title = "Scarlett Bangkok",
+                description = "We'd love to hear from you!",
+                coverImageUrl = "https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_"
+            ),
+            SurveyUiModel(
+                id = "2",
+                title = "ibis Bangkok Riverside",
+                description = "We'd love to hear from you!",
+                coverImageUrl = "https://dhdbhh0jsld0o.cloudfront.net/m/287db81c5e4242412cc0_"
+            ),
+            SurveyUiModel(
+                id = "3",
+                title = "21 on Rajah",
+                description = "We'd love to hear from you!",
+                coverImageUrl = "https://dhdbhh0jsld0o.cloudfront.net/m/0221e768b99dc3576210_"
+            )
+        )
+        currentDate = "TUESDAY, MAY 10"
+        appVersion = "v0.5.0"
+    }
+
+    HomeContentWithDrawer(
+        appVersion = appVersion,
+        scaffoldState = scaffoldState,
+        currentDate = currentDate,
+        isLoading = isLoading,
+        surveyUiModels = surveyUiModels,
+        userUiModel = userUiModel
+    )
+}
+
+@Composable
+private fun HomeContentWithDrawer(
+    appVersion: String,
+    initialDrawerState: DrawerValue = DrawerValue.Closed,
+    scaffoldState: ScaffoldState,
+    currentDate: String,
+    userUiModel: UserUiModel? = null,
+    surveyUiModels: List<SurveyUiModel>,
+    isLoading: Boolean,
+) {
+    val drawerState = rememberDrawerState(initialDrawerState)
+    val scope = rememberCoroutineScope()
+
+    RtlModalDrawer(
+        drawerState = drawerState,
+        gesturesEnabled = drawerState.isOpen,
+        drawerContent = {
+            HomeDrawer(
+                userUiModel = userUiModel,
+                appVersion = appVersion,
+                onLogoutClick = {
+                    scope.launch { drawerState.close() }
+                }
+            )
+        }
+    ) {
+        HomeContent(
+            scaffoldState = scaffoldState,
+            userUiModel = userUiModel,
+            currentDate = currentDate,
+            surveyUiModels = surveyUiModels,
+            isLoading = isLoading,
+            onUserAvatarClick = {
+                scope.launch { drawerState.open() }
+            }
+        )
+    }
 }
 
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
-fun HomeContent(modifier: Modifier = Modifier) {
-    Text(
-        text = "Home under construction",
-        style = MaterialTheme.typography.h1,
+fun HomeContent(
+    scaffoldState: ScaffoldState,
+    userUiModel: UserUiModel?,
+    isLoading: Boolean,
+    currentDate: String,
+    surveyUiModels: List<SurveyUiModel>,
+    onUserAvatarClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val pagerState = rememberPagerState()
```

Comment:
We can keep the `scaffoldState` here.
```suggestion
    val pagerState = rememberPagerState()
    val scaffoldState = rememberScaffoldState()
```